libtsuba/include/tsuba/ParquetReader.h

```diff
@@ -17,22 +23,67 @@ class KATANA_EXPORT ParquetReader {
     int64_t length;
   };
 
-  /// \returns a Reader that will read a table from storage location optionally
-  /// reading only part of the table defined by \param slice
+  struct ReadOpts {
+    /// if true (default) make sure canonical types are used and table columns
+    /// are not chunked
+    bool make_safe{true};
+
+    /// if provided, slice the resulting table so that it only contains
+    /// Slice.length rows starting from Slice.offset
+    std::optional<Slice> slice{std::nullopt};
+
+    static ReadOpts Defaults() { return ReadOpts{}; }
+  };
+
+  /// build a reader that will read a table from storage location optionally
+  /// reading only part of the table.
+  /// \param opts an opt structure detailing how reads should behave (see
+  ///    the ReadOpts struct definition for details)
   static katana::Result<std::unique_ptr<ParquetReader>> Make(
-      std::optional<Slice> slice = std::nullopt);
+      ReadOpts opts = ReadOpts::Defaults());
 
-  /// read table from \param uri
-  katana::Result<std::shared_ptr<arrow::Table>> ReadFromUri(
+  /// read table from storage
+  ///   \param uri an identifier for a parquet file
+  katana::Result<std::shared_ptr<arrow::Table>> ReadTable(
       const katana::Uri& uri);
 
+  /// read part of a table from storage
+  /// n.b. support for the `slice` read option is missing here
+  ///   \param uri an identifier for a parquet file
+  ///   \param column_bitmap must have the same length as the number of columns
+  ///      in the table in the parquet file. The loaded table will only contain
+  ///      columns at indexes that are true in the bitmap
+  katana::Result<std::shared_ptr<arrow::Table>> ReadTable(
+      const katana::Uri& uri, const std::vector<bool>& column_bitmap);
+
+  /// read a column part of a table from storage
+  /// n.b. support for the `slice` read option is missing here
+  ///   \param uri an identifier for a parquet file
+  ///   \param column_idx must be a valid column index for the table in that
+  ///      file
+  katana::Result<std::shared_ptr<arrow::Table>> ReadColumn(
+      const katana::Uri& uri, int32_t column_idx);
+
```

Comment:
This might be a philosophical issue, but I've taken the position in graphupdate that all properties are named by string.  I think this is most clear in the code and is most flexible when you might be reading in one set of columns and writing out another.  Also, given the limited number of properties we expect, maintaining the string names is not a drain on resources.

Reply:
This is for general reading of parquet not necessarily properties.